examples/tutorial/ArithmeticClient.hs

```diff
@@ -0,0 +1,38 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE OverloadedLists #-}
+{-# LANGUAGE GADTs #-}
+
+import Arithmetic
+import Network.GRPC.HighLevel.Generated
+
+clientConfig :: ClientConfig
+clientConfig = ClientConfig { clientServerHost = "localhost"
+                            , clientServerPort = 50051
+                            , clientArgs = []
+                            , clientSSLConfig = Nothing
+                            }
+
+main :: IO ()
+main = withGRPCClient clientConfig $ \client -> do
+  (Arithmetic arithmeticAdd arithmeticRunningSum) <- arithmeticClient client
+
+  -- Request for the Add RPC
+  ClientNormalResponse (OneInt x) _meta1 _meta2 _status _details
+    <- arithmeticAdd (ClientNormalRequest (TwoInts 2 2) 1 [])
+  putStrLn ("2 + 2 = " ++ show x)
+
+  -- Request for the RunningSum RPC
+  ClientWriterResponse reply _streamMeta1 _streamMeta2 streamStatus streamDtls
+    <- arithmeticRunningSum $ ClientWriterRequest 1 [] $ \send -> do
+        _ <- send (OneInt 1)
+        _ <- send (OneInt 2)
+        _ <- send (OneInt 3)
+        return ()
+
+  case reply of
+    Just (OneInt y) -> print ("1 + 2 + 3 = " ++ show y)
+    Nothing -> putStrLn ("Client stream failed with status "
```

Comment:
Does `Nothing` indicate failure? I'd think that `streamStatus` is a better indicator of success or failure (kind of like http's status code); you could match on that and for the successful or OK case then decompose the reply? Or do I have that backwards?

Elaborating on this (or tweaking it) might be useful for a newcomer like myself who isn't quite sure how to orient handling replies from the streaming API.